### PR TITLE
dask: clarification of `_Units` and `_hardmask` usage

### PR DESCRIPTION
**Pull request by davidhassell**

### CODE REVIEW

File: cf/data/data.py [hunk 1]
```diff
@@ -5696,7 +5696,25 @@ def _collapse_optimize_weights(self, weights):
...
+        The units are stored in a `Units` object, and reflect the
+        units of the (yet to be computed) elements of the underlying
+        data.
+
+        .. warning:: Assigning to `_Units` does *not* trigger a units
+                     conversion of the underlying data
+                     values. Therefore assigning to `_Units` should
+                     only be done in cases when it is known that the
+                     intrinsic units represented by the data values
+                     are inconsistent with the value existing value of
```
Comment (sadielbartholomew) [inline cf/data/data.py:5710]:
```suggestion
                     are inconsistent with the existing value of
```

File: cf/data/data.py [hunk 2]
```diff
@@ -5696,7 +5696,25 @@ def _collapse_optimize_weights(self, weights):
...
+                     `_Units`. Before assigning to `_Units`, first
+                     consider if assigning to `Units`, or calling the
+                     `override_units` or `override_calendar` method is
+                     a more approriate course of action, and use one
```
Comment (sadielbartholomew) [inline cf/data/data.py:5714]:
```suggestion
                     a more appropriate course of action, and use one
```

File: cf/data/data.py [hunk 3]
```diff
@@ -5758,6 +5776,18 @@ def _hardmask(self):
         Contains a `bool`, where `True` denotes a hard mask and
         `False` denotes a soft mask.
 
+        .. warning:: Assigning to `_hardmask` does *not* trigger a
+                     hardening orsoftening of the mask of the
```
Comment (sadielbartholomew) [inline cf/data/data.py:5780]:
```suggestion
                     hardening or softening of the mask of the
```

File: cf/data/data.py [hunk 4]
```diff
@@ -5758,6 +5776,18 @@ def _hardmask(self):
...
+                     underlying data values. Therefore assigning to
+                     `_hardmask` should only be done in cases when it
+                     is known that the intrinsic mask hardness of the
+                     data values is inconsistent with the value
```
Comment (sadielbartholomew) [inline cf/data/data.py:5784]:
```suggestion
                     data values is inconsistent with the
```

File: cf/data/data.py [hunk 5]
```diff
@@ -5758,6 +5776,18 @@ def _hardmask(self):
...
+                     existing value of `_hardmask`. Before assigning
+                     to `_hardmask`, first consider if assigning to
+                     `hardmask`, or calling the `harden_mask` or
+                     `soften_mask` method is a more approriate course
```
Comment (sadielbartholomew) [inline cf/data/data.py:5788]:
```suggestion
                     `soften_mask` method is a more appropriate course
```

**Review by sadielbartholomew — APPROVED** [review]
These warnings look very useful, thanks, and I will heed them for the Dask migration work going forwards. Some typos to fix, then please merge.

**Comment by davidhassell** [comment]
Thanks, Sadie (and apologies for my not checking the spelling and grammar sufficiently!).

**Comment by sadielbartholomew** [comment]
> apologies for my not checking the spelling and grammar sufficiently!

No worries at all, you are being very productive towards getting us through the big table of methods to migrate and it is very easy for a pair of second eyes to spot typos, which we all make anyway...

**Pull request merged**